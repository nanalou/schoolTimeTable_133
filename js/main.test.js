const { toShortTime, createOptions, cell } = require('./main');

describe("shorts the time-string", () => {
  test("createWeekCalculator function is given", () => {
    expect(toShortTime).toBeDefined();
  }),
  test("should short the time to only hours and min", () => {
    const time = '23:50:44';
    expect(toShortTime(time)).toBe('23:50');
  })
});

describe("create html component", () => {
  test("cell function is given", () => {
    expect(cell).toBeDefined();
  })
  test("cell-component renders correctly", () => {
    expect(cell('test')).toMatchSnapshot();
  })
  test("createOption function is given", () => {
    expect(createOptions).toBeDefined();
  })
  test("option-component renders correctly", () => {
    const data = [ 
      id => '42',
      name => 'testName'
    ];
    expect(createOptions(data, ['id', 'name'])).toMatchSnapshot();
  })
})
