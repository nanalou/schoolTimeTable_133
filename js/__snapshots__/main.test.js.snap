// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create cell-html component cell-component renders correctly 1`] = `"<td class=\\"border px-4 py-2\\">test</td>"`;

exports[`create html component cell-component renders correctly 1`] = `"<td class=\\"border px-4 py-2\\">test</td>"`;

exports[`create html component option-component renders correctly 1`] = `
Array [
  "
    <option value=\\"undefined\\">
      
    </option>",
  "
    <option value=\\"undefined\\">
      
    </option>",
]
`;

exports[`create option-html component option-component renders correctly 1`] = `
Array [
  "
    <option value=\\"undefined\\">
      
    </option>",
  "
    <option value=\\"undefined\\">
      
    </option>",
]
`;
